src/Symfony/Component/Cache/Adapter/FilesystemAdapter.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class FilesystemAdapter extends AbstractAdapter
+{
+    private $directory;
+
+    public function __construct($directory, $defaultLifetime = null)
+    {
+        parent::__construct('', $defaultLifetime);
+
+        $dir = rtrim($directory, '/'.DIRECTORY_SEPARATOR).DIRECTORY_SEPARATOR;
```

Comment:
Why do we use the `DIRECTORY_SEPARATOR` constant here (and below) to construct paths?